torch/_functorch/aot_autograd.py

```diff
@@ -1780,22 +1780,16 @@ def add_dupe_args(args):
     deduped_flat_args = remove_dupe_args(flat_args)
 
     tracing_context = TracingContext.get()
-    if tracing_context:
+    if tracing_context and len(tracing_context.aot_autograd_arg_pos_to_source) > 0:
```

Comment:
under what conditions can we not have any sources?

Reply:
This codepath is shared with all aot_auotgrad entry points, not just dynamo.